gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DiskStoreImpl.java

```diff
@@ -5168,6 +5171,12 @@ public boolean markIndexRecoveryScheduled() {
     }
   }
 
+  public void resetIndexRecoveryState() {
+    synchronized (this.indexRecoveryState) {
+      this.indexRecoveryState[0] = INDEXRECOVERY_UNINIT;
+    }
+  }
+
```

Comment:
Why is indexRecoveryState an array with size 1? Would be much cleaner to just make it AtomicInteger and get rid of the synchronization.